ch02.adoc

```diff
@@ -19,8 +19,9 @@ It is possible to treat the **`byte`** and **`short`** types as unsigned by usin
 In many situations, any integer type may be used.
 When the phrase "integer type" is used in this document, it should be understood to mean **`byte`**, **`unsigned byte`**, **`short`**, **`unsigned short`**, **`int`**, **`unsigned int`**, **`int64`**, or **`unsigned int64`**.
 
-A text string in a variable or an attribute may be represented either as Unicode text in a variable-length **`string`** or encoded as UTF-8 in a fixed-length **`char`** array.
-Note that the ASCII one-byte character codes (hexadecimal `00`-`7F`) are a subset of UTF-8.
+Text strings must be represented in Unicode. Any composite characters must be link:$$https://unicode.org/reports/tr15$$[NFC-normalized].
```

Comment:
There's still a language issue here -- "represented in Unicode" doesn't  have a precise meaning for binary data. In netcdf (and hdf) both `string` and array of `char` store bytes. Those bytes must be UTF-8 encoded. Both of these require an encoding. 

Netcdf and CF require the UTF-8 encoding.

Also "must be represented in Unicode" is correct, but might be confusing -- some folks *think* that Unicode means "not ASCII". 

How about:

> Text strings must be either ASCII or Unicode, encoded as UTF-8, in variable-length netCDF **`string`** or fixed-length **`char`** array.  Any composite characters must be link:$$https://unicode.org/reports/tr15$$[NFC-normalized]. Note that the ASCII character one-byte characters are a subset of Unicode, and their UTF-8 encodings are the same as their ASCII codes (decimal 0-127, hexadecimal `00`-`7F`).


The ASCII is redundant, but I think it might be helpful for us old timers that still don't quite "get" Unicode -- if someone is currently using ASCII, they can read this and know they are all good, and don't need to figure out what the heck UTF-8 means.

I'm not entirely sure about specifying "composite characters" in: 

"Any composite characters must be link:$$https://unicode.org/reports/tr15$$[NFC-normalized]"

as a rule, users don't need to think about what are or are not composite characters, they can apply the normalization to all strings, and that will be figured out for you. 

But it is correct , just not sure how best to be clear, but not confusing.